lib/proxy/http_download.rb

```diff
@@ -26,7 +26,7 @@ def initialize(src, dst, connect_timeout: DEFAULT_CONNECT_TIMEOUT, verify_server
       # keep last changed file attribute
       args << "--remote-time"
       # only download newer files
-      args += ["--time-cond", dst.to_s]
+      args += ["--time-cond", "file #{dst}"]
```

Comment:
Looking at the man page I wonder if it really is a single argument, or should be two. The manpage has these examples:
```
Examples:
 curl -z "Wed 01 Sep 2021 12:18:00" https://example.com
 curl -z "-Wed 01 Sep 2021 12:18:00" https://example.com
 curl -z file https://example.com
```

When I run it manually to simulate your version:
```console
$ curl -z "file initrd.img" https://mirror.stream.centos.org/9-stream/BaseOS/x86_64/os/isolinux/initrd.img -o initrd.img 
Warning: Illegal date format for -z, --time-cond (and not a file name). 
Warning: Disabling time condition. See curl_getdate(3) for valid date syntax.
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 95.3M  100 95.3M    0     0  90.7M      0  0:00:01  0:00:01 --:--:-- 90.8M
```

Which makes me think it should be:
```suggestion
      args += ["--time-cond", "file", dst.to_s]
```